watertap/costing/watertap_costing_package.py

```diff
@@ -61,23 +61,63 @@ def add_LCOW(self, flow_rate, name="LCOW"):
             name (optional) - name for the LCOW variable (default: LCOW)
         """
 
+        denominator = (
+            pyo.units.convert(flow_rate, to_units=pyo.units.m**3 / self.base_period)
+            * self.utilization_factor
+        )
+
         self.add_component(
             name,
             pyo.Expression(
                 expr=(
                     self.total_capital_cost * self.capital_recovery_factor
                     + self.total_operating_cost
                 )
-                / (
-                    pyo.units.convert(
-                        flow_rate, to_units=pyo.units.m**3 / self.base_period
-                    )
-                    * self.utilization_factor
-                ),
+                / denominator,
                 doc=f"Levelized Cost of Water based on flow {flow_rate.name}",
             ),
         )
 
+        c_units = self.base_currency
+        t_units = self.base_period
+        unit_lcows = pyo.Expression(
+            pyo.Any,
+            doc=f"Levelized Cost of Water per unit based on flow {flow_rate.name}",
+        )
+        self.add_component(name + "_per_unit", unit_lcows)
+        for u in self._registered_unit_costing:
+            numerator = 0
+            if hasattr(u, "capital_cost"):
+                capital_cost = pyo.units.convert(u.capital_cost, to_units=c_units)
+                # total capital costs w/ recovery factor
+                numerator += self.capital_recovery_factor * (
+                    self.total_investment_factor * capital_cost
+                )
+                # maintenance_labor_chemical_operating_cost,
+                # part of total_fixed_operating_cost
+                numerator += self.maintenance_labor_chemical_factor * capital_cost
+            if hasattr(u, "fixed_operating_cost"):
+                numerator += pyo.units.convert(
+                    u.fixed_operating_cost, to_units=c_units / t_units
+                )
+            if hasattr(u, "variable_operating_cost"):
+                numerator += pyo.units.convert(
+                    u.variable_operating_cost, to_units=c_units / t_units
+                )
+            unit_lcows[u.unit_model.name] = numerator / denominator
+
+        flow_lcows = pyo.Expression(
+            pyo.Any,
+            doc=f"Levelized Cost of Water per flow based on flow {flow_rate.name}",
+        )
+        self.add_component(name + "_per_flow", flow_lcows)
+        for ftype, flows in self._registered_flows.items():
+            cost_var = getattr(self, f"{ftype}_cost")
+            for f in flows:
+                # part of total_variable_operating_cost
+                flow_cost = pyo.units.convert(f * cost_var, to_units=c_units / t_units)
+                flow_lcows[str(f)] = (flow_cost * self.utilization_factor) / denominator
```

Comment:
I suppose we should be aware of a potential edge case--where the primary treatment segment of a plant may be turned off, but there is still some minimum load for parts of the plant that stay on. Up until now, we've assumed that there is absolutely no electricity usage during shutdown when in fact, while we may not be producing treated water at a given time, we can still be, e.g., pretreating water with units that must stay on even when the primary treatment step shuts down.

Reply:
They'll get an error when evaluating this (and the other expressions) then. Not sure there's much to be done about it.